scripts/test/Muon/PeriodicTablePresenter_test.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import, print_function
+
+import unittest
+
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_presenter import PeriodicTablePresenter
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_view import PeriodicTableView
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_model import PeriodicTableModel
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table import PeriodicTable as silxPT
+
+from Muon.GUI.Common import mock_widget
+
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+
+class PeriodicTablePresenterTest(unittest.TestCase):
+    def setUp(self):
+        self._qapp = mock_widget.mockQapp()
+        self._model = mock.create_autospec(PeriodicTableModel)
+        self.presenter = PeriodicTablePresenter(
+            PeriodicTableView(), self._model)
+        self.mock_elem = mock.Mock()
+
+        self.presenter.view.ptable = mock.create_autospec(silxPT)
+        self.presenter.view.ptable.getSelection = mock.Mock(
+            return_value=self.mock_elem)
+        self.presenter.view.ptable.isElementSelected = mock.Mock(
+            return_value=True)
+        self.presenter.view.ptable.setSelection = mock.Mock()
+        self.presenter.view.ptable.setElementSelected = mock.Mock()
+
+        self.presenter.is_selected = mock.Mock()
+
+        self.presenter.view.on_table_lclicked = mock.Mock()
+        self.presenter.view.on_table_rclicked = mock.Mock()
+        self.presenter.view.on_table_changed = mock.Mock()
+
+        self.presenter.view.unreg_on_table_lclicked = mock.Mock()
+        self.presenter.view.unreg_on_table_rclicked = mock.Mock()
+        self.presenter.view.unreg_on_table_changed = mock.Mock()
```

Comment:
it would be neater to create the mocks on ```self.view``` then set the presenter to use the mocked view